### PR TITLE
feat(ecs-patterns): enable assignPublicIp on NLB Fargate Services by default

### PR DESCRIPTION
**Pull request by pahud**
feat(ecs-patterns): public-facing NLB fronted ECS tasks on Fargate with assignPublicIp enabled
 by defaults to allow all ingress traffic to the container port.

Seperate a new PR  from this [PR](https://github.com/aws/aws-cdk/pull/4825#issuecomment-549494627).

@pkandasamy91 please kindly review this PR.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
@SoManyHs  @pkandasamy91  can you kindly review this PR for me? Thanks.

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts [hunk 1]
```diff
@@ -135,5 +140,10 @@ export class NetworkLoadBalancedFargateService extends NetworkLoadBalancedServic
       cloudMapOptions: props.cloudMapOptions,
     });
     this.addServiceAsTarget(this.service);
+    // All public-facing NLB fronted fargate service with assignPublicIp by defaults allow from any IPv4 traffic
+    const publicNLB = props.publicLoadBalancer !== undefined ? props.publicLoadBalancer : true;
+    if (publicNLB && props.assignPublicIp) {
+      this.service.connections.connections.allowFromAnyIpv4(Port.tcp(containerPort));
```
Comment (pkandasamy91) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts]: This is not necessarily true, and it would be a breaking change. I think we should rely on users to determine if they want to allow the port or not.

File: packages/@aws-cdk/aws-ecs-patterns/test/fargate/test.load-balanced-fargate-service.ts [hunk 2]
```diff
@@ -29,6 +29,35 @@ export = {
     test.done();
   },
 
+  'setting public-facing NLB with fargate with assignPublicIp creates securigy group allow all'(test: Test) {
```
Comment (SoManyHs) [inline packages/@aws-cdk/aws-ecs-patterns/test/fargate/test.load-balanced-fargate-service.ts]: nit: typo - security

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts [hunk 3]
```diff
@@ -135,5 +140,10 @@ export class NetworkLoadBalancedFargateService extends NetworkLoadBalancedServic
       cloudMapOptions: props.cloudMapOptions,
     });
     this.addServiceAsTarget(this.service);
+    // All public-facing NLB fronted fargate service with assignPublicIp by defaults allow from any IPv4 traffic
```
Comment (SoManyHs) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts]: Can you rephrase this comment to be clearer?

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts (hunk 1 above)
Comment (SoManyHs) [reply packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts]: @pkandasamy91 since `containerPort` is being set in [line 128](https://github.com/aws/aws-cdk/pull/4856/files#diff-34db4be841bead71ec835c013f79f8a9R128), this effectively is allowing the customer to determine which port the NLB is receiving traffic on, no? Would this still be a breaking change?

File: packages/@aws-cdk/aws-ecs-patterns/test/fargate/test.load-balanced-fargate-service.ts [hunk 4]
```diff
@@ -29,6 +29,35 @@ export = {
...
+      cluster,
+      publicLoadBalancer: true,
+      assignPublicIp: true,
+      taskImageOptions: {
+        image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+      },
+    });
+
+    // THEN
+    expect(stack).to(haveResourceLike('AWS::EC2::SecurityGroup', {
```
Comment (SoManyHs) [inline packages/@aws-cdk/aws-ecs-patterns/test/fargate/test.load-balanced-fargate-service.ts]: Apologies if I'm missing something, but would we not want to assert on the SecurityGroupIngress rather than Egress?

**Comment by pahud** [comment]
Sorry I am closing this issue for now. Will get back and re-open it when necessary.
